app/views/planning_applications/edit_constraints_form.html.erb

```diff
@@ -0,0 +1,127 @@
+<% add_parent_breadcrumb_link "Home", planning_applications_path %>
+<% add_parent_breadcrumb_link "Application", planning_application_path(@planning_application) %>
+<% content_for :title, "Edit constraints" %>
+
+<%= render "planning_applications/proposal_header" %>
+
+<h2 class="govuk-heading-m">Constraints</h2>
+
+<%= form_with model: @planning_application, url: edit_constraints_planning_application_path(@planning_application), local: true do |form| %>
+
+
+  <div class="govuk-grid-column-two-thirds">
+
+    <div class="govuk-grid-row govuk-!-margin-top-6">
+      <h3 class="govuk-heading-s">General Policy</h3>
+      <div class="form-group">
+        <div class="govuk-checkboxes govuk-checkboxes--small">
+          <%= form.collection_check_boxes :constraints, policy_constraints, :itself, :itself do |b| %>
+            <div class="govuk-checkboxes__item">
+              <%= b.check_box class: "govuk-checkboxes__input" %>
+              <%= b.label class: "govuk-label govuk-checkboxes__label tag_checkbox_label" %>
+            </div>
+          <% end %>
+        </div>
+      </div>
+    </div>
+
+    <div class="govuk-grid-row govuk-!-margin-top-6">
+      <div class="form-group">
+        <h3 class="govuk-heading-s">Heritage & Conservation</h3>
+        <div class="govuk-checkboxes govuk-checkboxes--small">
+          <%= form.collection_check_boxes :constraints, heritage_constraints, :itself, :itself do |b| %>
+            <div class="govuk-checkboxes__item">
+              <%= b.check_box class: "govuk-checkboxes__input" %>
+              <%= b.label class: "govuk-label govuk-checkboxes__label tag_checkbox_label" %>
+            </div>
+          <% end %>
+        </div>
+      </div>
+    </div>
```

Comment:
As this is repeatedly duplicated code, I'm wondering if we could DRY it up a bit. Something like:

```erb
<% {
   "Heritage & Conservation" => heritage_constraints,
...
}.each do |constraint_group_title, constraints_array| %>
...
<h3 class="govuk-heading-s"><%= title %></h3>
...
<%= form.collection_check_boxes :constraints, constraints_array, :itself, :itself do |b| %>
...
<% end %>
```

Reply:
Good idea - will implement this.

Reply:
Done - it's much better.